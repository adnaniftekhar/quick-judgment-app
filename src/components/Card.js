import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export const Card = ({ data }) => {
  return (
    <View style={styles.card}>
      <Text style={styles.text}>
        {data?.statement || 'No statement provided'}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: 'white',
    borderRadius: 12,
    padding: 30,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 3,
    },
    shadowOpacity: 0.3,
    shadowRadius: 4.84,
    elevation: 7,
    minHeight: 250,
    width: '90%',
    alignSelf: 'center',
    justifyContent: 'center',
  },
  text: {
    fontSize: 24,
    textAlign: 'center',
    lineHeight: 32,
    fontWeight: '500',
    color: '#333',
  }
}); 