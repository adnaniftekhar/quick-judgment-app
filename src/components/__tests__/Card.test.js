import React from 'react';
import renderer from 'react-test-renderer';

// Mock react-native with StyleSheet support
jest.mock('react-native', () => ({
  View: 'View',
  Text: 'Text',
  StyleSheet: {
    create: styles => styles
  }
}));

import Card from '../Card';

describe('Card Component', () => {
  it('renders correctly with styles', () => {
    const tree = renderer.create(
      <Card data={{ statement: 'Test Statement' }} />
    ).toJSON();
    
    expect(tree.props.style).toEqual({
      padding: 15,
      backgroundColor: '#fff',
      borderRadius: 8,
      shadowColor: '#000',
      shadowOpacity: 0.1,
      shadowRadius: 4,
      elevation: 2,
    });

    expect(tree.children[0].props.style).toEqual({
      fontSize: 16,
      color: '#333',
    });
  });

  it('matches snapshot with styles', () => {
    const tree = renderer.create(
      <Card data={{ statement: 'Snapshot test' }} />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('handles empty statement', () => {
    const tree = renderer.create(
      <Card data={{ statement: '' }} />
    ).toJSON();
    
    // More defensive check - verify the structure exists first
    expect(tree.type).toBe('View');
    expect(tree.children[0].type).toBe('Text');
    // For empty string, the Text component might have no children or an empty string
    const textContent = tree.children[0].children ? tree.children[0].children[0] : '';
    expect(textContent).toBe('');
  });

  it('handles long statement', () => {
    const longText = 'This is a very long statement that might need special handling in the UI';
    const tree = renderer.create(
      <Card data={{ statement: longText }} />
    ).toJSON();
    expect(tree.children[0].children[0]).toBe(longText);
  });

  it('handles missing data prop gracefully', () => {
    const consoleError = console.error;
    console.error = jest.fn(); // Suppress React prop type warnings in test output
    
    const tree = renderer.create(
      <Card />
    ).toJSON();
    
    expect(tree.type).toBe('View');
    expect(tree.children[0].type).toBe('Text');
    const textContent = tree.children[0].children ? tree.children[0].children[0] : '';
    expect(textContent).toBe('');
    
    console.error = consoleError; // Restore console.error
  });
}); 