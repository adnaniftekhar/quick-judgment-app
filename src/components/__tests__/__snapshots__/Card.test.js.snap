// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot with styles 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 8,
      "elevation": 2,
      "padding": 15,
      "shadowColor": "#000",
      "shadowOpacity": 0.1,
      "shadowRadius": 4,
    }
  }
  testID="card-container"
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 16,
      }
    }
    testID="card-text"
  >
    Snapshot test
  </Text>
</View>
`;
